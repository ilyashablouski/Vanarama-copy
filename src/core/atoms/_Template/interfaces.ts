import { IBaseProps } from '../../interfaces/base';

export interface ITemplateProps extends IBaseProps {}
