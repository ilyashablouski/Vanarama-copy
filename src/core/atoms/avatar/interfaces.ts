import { IBaseProps } from '../../interfaces/base';
import { TSize } from '../../../types/size';

export interface IAvatarProps extends IBaseProps {
  src?: string;
  size?: TSize;
  altText?: string;
}
