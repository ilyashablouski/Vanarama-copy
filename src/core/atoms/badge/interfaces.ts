import { IBaseProps } from '../../interfaces/base';
import { TColor } from '../../../types/color';
import { TSize } from '../../../types/size';

export interface IBadgeProps extends IBaseProps {
  size?: TSize;
  color?: TColor;
  label?: string;
}
