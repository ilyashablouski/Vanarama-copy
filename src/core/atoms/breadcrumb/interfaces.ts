import { IBaseProps } from '../../interfaces/base';
import { ILink } from '../../interfaces/link';

export interface IBreadcrumbProps extends IBaseProps {
  items?: ILink[];
}
