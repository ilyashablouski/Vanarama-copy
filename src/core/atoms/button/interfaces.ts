import { ButtonHTMLAttributes } from 'react';
import { IBaseProps } from '../../interfaces/base';
import { TColor } from '../../../types/color';
import { TSize } from '../../../types/size';

export interface IButtonProps
  extends IBaseProps,
    ButtonHTMLAttributes<HTMLButtonElement> {
  color?: TColor;
  fill?: 'solid' | 'outline' | 'clear';
  icon?: React.ReactNode;
  iconColor?: TColor;
  iconPosition?: 'before' | 'after';
  label?: string | React.ReactNode;
  round?: boolean;
  size?: TSize;
  withoutDefaultClass?: boolean;
  customCTAColor?: string;
}
