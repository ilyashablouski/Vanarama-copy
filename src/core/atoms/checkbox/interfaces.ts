import { InputHTMLAttributes, ReactElement } from 'react';
import { IBaseProps } from '../../interfaces/base';
import { TColor } from '../../../types/color';

export interface ICheckboxProps
  extends IBaseProps,
    InputHTMLAttributes<HTMLInputElement> {
  /**
   * A color for the checkbox.
   */
  color?: TColor;
  /**
   * A unique ID for the checkbox node - this is required here because the
   * underlying checkbox is hidden and therefore you can only click on the label
   * and the label needs to know which input it corresponds to.
   */
  id: string;
  /**
   * 	Invalid state of the checkbox.
   */
  invalid?: boolean;
  /**
   * A name for the input node.
   */
  label: string | (string | ReactElement)[];
  /**
   * Outlined state of the checkbox.
   */
  outline?: boolean;
  /**
   * A ref to attach to the checkbox.
   */
  ref?: React.Ref<HTMLInputElement>;
}
