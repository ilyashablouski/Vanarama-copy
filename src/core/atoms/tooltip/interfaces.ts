import { IBaseProps } from '../../interfaces/base';
import { TColor } from '../../../types/color';

export interface ITooltipProps extends IBaseProps {
  color?: TColor;
  text: string;
  icon?: React.ReactNode;
  position?: string;
}
