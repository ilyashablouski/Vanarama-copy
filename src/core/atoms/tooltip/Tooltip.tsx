import React, { FC, memo } from 'react';
import cx from 'classnames';

import Icon from '../icon';

import InformationCircleSharp from '../../assets/icons/InformationCircleSharp';

import { ITooltipProps } from './interfaces';

const Tooltip: FC<ITooltipProps> = memo(props => {
  const {
    className,
    color = 'teal',
    icon = <InformationCircleSharp />,
    position = 'top center',
    text,
  } = props;

  return (
    <span
      className={cx('tooltip', className, {
        [`-${color}`]: color,
        [position]: position,
      })}
    >
      <Icon icon={icon} color={color} />
      <span className="tooltiptext" data-tooltip={text} />
    </span>
  );
});

Tooltip.displayName = 'Tooltip';

export default Tooltip;
