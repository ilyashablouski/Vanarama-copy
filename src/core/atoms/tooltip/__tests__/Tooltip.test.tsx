import React from 'react';
import renderer from 'react-test-renderer';

import Tooltip from '..';
import { ITooltipProps } from '../interfaces';

const optionalProps = {
  id: 'component',
  className: 'class',
  color: 'teal' as const,
  position: 'bottom left',
};

const mandatoryProps = {
  text: 'Text text text text text',
};

function getComponent(props?: Partial<ITooltipProps>) {
  return renderer
    .create(<Tooltip text={mandatoryProps.text} {...props} />)
    .toJSON();
}

describe('<Tooltip />', () => {
  it('renders correctly', () => {
    const tree = getComponent();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with optional props', () => {
    const tree = getComponent(optionalProps);
    expect(tree).toMatchSnapshot();
  });
});
