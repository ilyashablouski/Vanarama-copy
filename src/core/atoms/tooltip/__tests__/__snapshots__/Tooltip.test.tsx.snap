// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> renders correctly 1`] = `
<span
  className="tooltip -teal top center"
>
  <i
    className="icon -teal"
  >
    <svg
      height="1em"
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M256,56C145.72,56,56,145.72,56,256s89.72,200,200,200,200-89.72,200-200S366.28,56,256,56Zm0,82a26,26,0,1,1-26,26A26,26,0,0,1,256,138Zm64,226H200V332h44V244H212V212h64V332h44Z"
      />
    </svg>
  </i>
  <span
    className="tooltiptext"
    data-tooltip="Text text text text text"
  />
</span>
`;

exports[`<Tooltip /> renders correctly with optional props 1`] = `
<span
  className="tooltip class -teal bottom left"
>
  <i
    className="icon -teal"
  >
    <svg
      height="1em"
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M256,56C145.72,56,56,145.72,56,256s89.72,200,200,200,200-89.72,200-200S366.28,56,256,56Zm0,82a26,26,0,1,1-26,26A26,26,0,0,1,256,138Zm64,226H200V332h44V244H212V212h64V332h44Z"
      />
    </svg>
  </i>
  <span
    className="tooltiptext"
    data-tooltip="Text text text text text"
  />
</span>
`;
